com.unity.multiplayer.mlapi/Runtime/Messaging/MessageQueue/MessageQueueProcessor.cs

```diff
@@ -325,12 +337,21 @@ private void SendCallback(ulong clientId, MessageBatcher.SendStream sendStream)
         /// <param name="item">Information on what to send</param>
         private void SendFrameQueueItem(MessageFrameItem item)
         {
+            var channel = item.NetworkChannel;
+            // If the length is greater than the fragmented threshold, switch to a fragmented channel.
+            // This is kind of a hack to get around issues with certain usages patterns on fragmentation with UNet.
```

Comment:
I'm curious is there a Slack thread or something to learn more about that UNet fragmentation issue?

Reply:
No slack thread. But I'll tell you the behavior we see...

With UNet delivering things using ReliableFragmentedSequenced and using batched messages at around 530 bytes per batch, UNet runs out of buffer space in SceneTransitioningTest with as little as 6 objects spawning per second and starts dropping packets.

If we turn off batching - sending the same amount of overall data but in smaller 40 byte chunks instead of combined 530 byte chunks - it can get up to ~80 per second before packtes start getting dropped.

If we send on ReliableSequenced, it got up to 550 per second (the maximum the slider would allow) and nothing got dropped.

So 1) Fragmented is dropping messages that the network can handle without issue, and 2) for some reason sending the same data in fewer, larger chunks causes it to drop them more aggressively than sending in more, smaller chunks, even with the same amount of overall bandwidth usage.

I don't know where to find the UNet source to try to better understand why that happens, but I think we should avoid using ReliableFragmentedSequenced as much as possible.

Reply:
I guess that kind of makes sense. Maybe you could raise this in the internal unet slack channel just so that the right people know that this is an existing issue. And maybe they even have a fix for it. Ignore this if you already did.

Reply:
What is the internal unet channel? I will go raise it there.